nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/main/java/org/apache/nifi/processors/smb/ListSmb.java

```diff
@@ -0,0 +1,342 @@
+
+
+    public static final PropertyDescriptor SMB_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(LISTING_STRATEGY)
+            .allowableValues(BY_ENTITIES, NO_TRACKING, BY_TIMESTAMPS)
+            .build();
+
+    public static final PropertyDescriptor SMB_CONNECTION_POOL_SERVICE = new Builder()
+            .name("smb-client-provider-service")
+            .displayName("SMB Client Provider Service")
+            .description("Specifies the SMB client provider to use for creating SMB connections.")
+            .required(true)
+            .identifiesControllerService(SmbClientProviderService.class)
+            .build();
+
+    public static final PropertyDescriptor SKIP_FILES_WITH_SUFFIX = new Builder()
+            .name("file-name-suffix-filter")
+            .displayName("File Name Suffix Filter")
+            .description("Files ends with the given suffix will be omitted. This is handy when writing large data into "
+                    + "temporary files and then moved to a final one. Please be advised that writing data into files "
+                    + "first is highly recommended when using Entity Tracking or Timestamp based listing strategies.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .addValidator(new MustNotContainDirectorySeparatorsValidator())
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = unmodifiableList(asList(
+            SMB_LISTING_STRATEGY,
+            SMB_CONNECTION_POOL_SERVICE,
+            DIRECTORY,
+            AbstractListProcessor.RECORD_WRITER,
+            SKIP_FILES_WITH_SUFFIX,
+            MINIMUM_AGE,
+            MINIMUM_SIZE,
+            MAXIMUM_SIZE,
+            AbstractListProcessor.TARGET_SYSTEM_TIMESTAMP_PRECISION,
+            ListedEntityTracker.TRACKING_STATE_CACHE,
+            ListedEntityTracker.TRACKING_TIME_WINDOW,
+            ListedEntityTracker.INITIAL_LISTING_TARGET
+    ));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    protected Map<String, String> createAttributes(SmbListableEntity entity, ProcessContext context) {
+        final Map<String, String> attributes = new TreeMap<>();
+        attributes.put("filename", entity.getName());
+        attributes.put("shortname", entity.getShortName());
+        attributes.put("path", entity.getPath());
+        attributes.put("absolute.path", getPath(context) + entity.getPathWithName());
+        attributes.put("identifier", entity.getIdentifier());
+        attributes.put("timestamp",
+                ISO_DATE_TIME.format(LocalDateTime.ofEpochSecond(entity.getTimestamp(), 0, ZoneOffset.UTC)));
+        attributes.put("creationTime",
+                ISO_DATE_TIME.format(LocalDateTime.ofEpochSecond(entity.getCreationTime(), 0, ZoneOffset.UTC)));
+        attributes.put("lastAccessedTime",
```

Comment:
Suggestion: `lastAccessedTime` => `lastAccessTime` (as in case of the record writer approach)